### DISCUSSION TITLE
Duck Db Creating 100 Gb of Temp file

### DISCUSSION THREAD

**Discussion by akash-goel** [duckdb/duckdb#6021] [General]
Hi Team,

When we are converting Big data file ( 7 GB .duckdb file ) to Parquet format than duck db is creating the tmp file duckdb_temp_storage-0.tmp in 100GB , Can you please suggest how we can resolve this issue.

**Comment by Mytherin** [comment]
That is unexpected. The parquet conversion should be done in streaming fashion - so only individual row groups have to be materialized. How many rows are in the database, and what are the data types of the rows?

**Reply by akash-goel** [reply]
Hi Mytherin ,

we have 64 million rows and 500 columns . DataType is Varchar for all the fields .  I am using the 32 GB RAM system  to convert the duck db table into parquet file

Below Code I am using to convert , where test_df is the table in the database which contains 64 million records.
#con.execute("COPY test_df TO 'result-snappy.parquet' (FORMAT 'PARQUET', CODEC 'SNAPPY')")

Regards,
Akash
